### PR TITLE
feat: add status check for Restore

### PR DESCRIPTION
**Pull request by qbiqing**
Add option to check the status of restore for an object.

Commonize object listing into a non-Glacier specific function.

### CODE REVIEW

**Comment by qbiqing** [comment]
Tested the new function on cloud

File: s3-glacier-restore.py [hunk 1]
```diff
@@ -89,13 +94,16 @@ def list_glacier_objects(
             Prefix=prefix,
             **kwargs,
         )
-
         if "Contents" in rsp:
             for obj in rsp["Contents"]:
-                if obj["StorageClass"] == "GLACIER":
-                    key = obj["Key"]
+                key = obj["Key"]
+                if glacier == True:
```
Comment (guangie88) [inline s3-glacier-restore.py]: Nit: `if glacier` would do

**Review by guangie88 — APPROVED** [review]
LGTM with a trivial nitpick!

File: s3-glacier-restore.py (hunk 1 above)
Comment (qbiqing) [reply s3-glacier-restore.py]: edited: https://github.com/guangie88/s3-glacier-restore/pull/3/commits/f9c5027ff6c0b0e0c67fded78547173ff6c4edf7

**Pull request merged**